app/Http/Controllers/AjaxController.php

```diff
@@ -29,4 +29,31 @@ public function addfriend(Request $req) {
         }
         return response()->json(array("message" => "generic error message"), 500);
     }
+
+    /*
+     * AJAX Request to handle accepting/declining a friend request
+     *
+     */
+    public function processFriendRequest(Request $req) {
+        if ($req->json() && isset ($req->username)) {
+            $friendship = Friend::whereIn('username1', [Auth::user()->username, $req->username])
+                ->whereIn('username2', [Auth::user()->username, $req->username])
+                ->where('status_code', 'pending')
+                ->get();
+            if (isset ($req->accepted)) {
+                if ($req->accepted == 'accepted') {
+                    $friendship->status_code = 'accepted';
+                    return response()->json(array("message" => "friend request accepted!"), 200);
+                } else if ($req->accepted == 'denied') {
+                    $friendship->status_code = 'denied';
+                    return response()->json(array("message" => "friend request denied!"), 200);
+                }
+                $friendship->save(); //may not actually save into db
```

Comment:
you should have the return statement after this line or else you wont be able to save